editing/add-features-with-contingent-values/src/main/java/com/esri/samples/add_features_with_contingent_values/AddFeaturesWithContingentValuesController.java

```diff
@@ -0,0 +1,437 @@
+import com.esri.arcgisruntime.mapping.Basemap;
+import com.esri.arcgisruntime.mapping.view.Graphic;
+import com.esri.arcgisruntime.mapping.view.GraphicsOverlay;
+import com.esri.arcgisruntime.mapping.view.MapView;
+import com.esri.arcgisruntime.symbology.ColorUtil;
+import com.esri.arcgisruntime.symbology.SimpleFillSymbol;
+import com.esri.arcgisruntime.symbology.SimpleLineSymbol;
+import com.esri.arcgisruntime.symbology.SimpleMarkerSymbol;
+import com.esri.arcgisruntime.symbology.SimpleRenderer;
+
+public class AddFeaturesWithContingentValuesController {
+
+  @FXML private MapView mapView;
+  @FXML private ComboBox<CodedValue> statusComboBox;
+  @FXML private ComboBox<CodedValue> protectionComboBox;
+  @FXML private Label label;
+  @FXML private Slider bufferSlider;
+  @FXML private VBox attributeControlsVBox;
+  @FXML private Label addFeatureLabel;
+  @FXML private Label attributesLabel;
+
+  private ArcGISFeature newFeature;
+  private ContingentValuesDefinition contingentValuesDefinition;
+  private FeatureLayer featureLayer;
+  private Graphic graphic;
+  private Geodatabase geodatabase;
+  private GeodatabaseFeatureTable geodatabaseFeatureTable;
+  private GraphicsOverlay graphicsOverlay;
+  private QueryParameters queryParameters;
+
+  private final SimpleBooleanProperty isEditingFeatureProperty = new SimpleBooleanProperty(false);
+
+  /**
+   * Called after FXML loads. Sets up map.
+   */
+  public void initialize() {
+
+    try {
+
+      // authentication with an API key or named user is required to access basemaps and other location services
+      String yourAPIKey = System.getProperty("apiKey");
+      ArcGISRuntimeEnvironment.setApiKey(yourAPIKey);
+
+      // display the name of the coded values in the attribute combo boxes
+      protectionComboBox.setButtonCell(new CodedValueListCell());
+      protectionComboBox.setCellFactory(c -> new CodedValueListCell());
+      statusComboBox.setButtonCell(new CodedValueListCell());
+      statusComboBox.setCellFactory(c -> new CodedValueListCell());
+
+      // control updates to the UI
+      isEditingFeatureProperty.addListener(((observable, oldValue, newValue) -> {
+        if (newValue) {
+          addFeatureLabel.setVisible(false);
+          attributesLabel.setVisible(true);
+          attributeControlsVBox.setDisable(false);
+        } else {
+          addFeatureLabel.setVisible(true);
+          attributesLabel.setVisible(false);
+          attributeControlsVBox.setDisable(true);
+        }
+      }));
+
+      // create a new vector tile package from a vector tile package path
+      var vectorTiledLayer = new ArcGISVectorTiledLayer(new File(System.getProperty("data.dir"), "./samples-data" +
+              "/FillmoreTopographicMap.vtpk").getAbsolutePath());
+      // create a new basemap with the vector tiled layer and create a new map from it
+      var basemap = new Basemap(vectorTiledLayer);
+      ArcGISMap map = new ArcGISMap(basemap);
```

Comment:
could be a var?

Reply:
I prefer to not use `var` with this, purely because `map` could really mean anything in a Java application. Explicitly using `ArcGISMap` helps readability in this case. We could use `arcGISMap` as the variable name with `var`, but that kind of goes against the readability too.